helix-term/src/ui/editor.rs

```diff
@@ -830,6 +833,49 @@ impl EditorView {
         }
     }
 
+    /// Apply the highlighting on the columns where a cursor is active
+    pub fn highlight_cursorcolumn(
+        doc: &Document,
+        view: &View,
+        surface: &mut Surface,
+        theme: &Theme,
+    ) {
+        let text = doc.text().slice(..);
+
+        let primary_style = theme
+            .try_get("ui.cursorcolumn.primary")
+            .unwrap_or_else(|| theme.get("ui.cursorline.primary"));
```

Comment:
For the fallback behavior, I think we should check `ui.cursorcolumn.primary` first, then `unwrap_or_else` to check `ui.cursorcolumn`, then finally unwrap_or_else to `theme.get("ui.cursorline.primary")` (and same for secondary). That way we emulate the natural fallback behavior